base/src/schema.test.ts

```diff
@@ -915,6 +927,213 @@ describe('schema', () => {
 
       expect(() => Schema.validateSchema(schema)).not.toThrow()
     })
+
+    it('throws an error when schema definitions is not an object', () => {
+      const schema = [{ stuffs: { type: 'array' } }]
+
+      expect(() =>
+        Schema.validateSchema(schema as unknown as SchemaDefinitions)
+      ).toThrow('schema has to be an object')
+    })
+
+    it('throws a validation error when type is unknown', () => {
+      const schema = {
+        stuffs: { type: '' },
+      }
+
+      expect(() =>
+        Schema.validateSchema(schema as unknown as SchemaDefinitions)
+      ).toThrow('`type` is invalid or undefined')
+    })
+
+    it('throws a validation error when items isnt defined for an array', () => {
```

Comment:
```suggestion
    it('throws a validation error when items isn\'t defined for an array', () => {
```

Perhaps, group the schema validation into their types

`describe('array type', ...)`
`describe('object type', ...)`